pkg/virt-manifest/rest/manifest.go

```diff
@@ -94,11 +95,13 @@ func (m manifest) mapManifest(_ context.Context, request interface{}) (interface
 
 	virt_manifest.AddMinimalVMSpec(vm)
 
-	mappedVm, err := virt_manifest.MapVM(m.connection, vm)
-	mappedVm.Spec.Domain.Name = mappedVm.ObjectMeta.Name
+	client, err := kubecli.Get()
+
+	mappedVm, err := virt_manifest.MapVM(m.connection, vm, client)
```

Comment:
This client connection is something we wanted to avoid, for mapping the code in this service. PVCs are not necessarily even there when this one is called, that is why it was mocked out.